libevm/VMFactory.cpp

```diff
@@ -24,43 +24,115 @@
 #include "SmartVM.h"
 #endif
 
+namespace po = boost::program_options;
+
 namespace dev
 {
 namespace eth
 {
 namespace
 {
-	auto g_kind = VMKind::Interpreter;
+auto g_kind = VMKind::Interpreter;
+
+/// A helper type to build the tabled of VM implementations.
+///
+/// More readable than std::tuple.
+/// Fields are not initialized to allow usage of construction with initializer lists {}.
+struct VMKindTableEntry
+{
+    VMKind kind;
+    const char* name;
+};
+
+/// The table of available VM implementations.
+///
+/// We don't use a map to avoid complex dynamic initialization. This list will never be long,
+/// so linear search only to parse command line arguments is not a problem.
+VMKindTableEntry vmKindsTable[] = {
+    {VMKind::Interpreter, "interpreter"},
+#if ETH_EVMJIT
+    {VMKind::JIT, "jit"},
+    {VMKind::Smart, "smart"},
+#endif
+};
+}
+
+void validate(boost::any& v, const std::vector<std::string>& values, VMKind* /* target_type */, int)
```

Comment:
Where is this used?
And where is now getting `VMKind` from option string value?

Reply:
This overload is required by `po::value<VMKind>`. This function also coverts the string to `VMKind`.